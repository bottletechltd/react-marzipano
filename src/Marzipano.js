import React, { useRef } from 'react'
import styled from 'styled-components'

import useMarzipano from './useMarzipano'


const Root = styled.div`
  width: 100%;
  height: 100%;
`

const ViewerCanvas = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
`

export default function Marzipano(props) {
  const viewerCanvas = useRef(null)
  useMarzipano(viewerCanvas, props)

  return (
    <Root>
      <ViewerCanvas ref={viewerCanvas} />
    </Root>
  )
}
