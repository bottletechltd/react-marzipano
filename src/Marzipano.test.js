import React from 'react'
import renderer from 'react-test-renderer'

import Marzipano from './Marzipano'


test('Match snapshot', () => {
  const marzipano = renderer.create(
    (<Marzipano scenes={[{
      current: true,
      imageUrl: 'https://www.marzipano.net/media/equirect/angra.jpg',
      type: 'equirect'
    }]} />)
  )
  const tree = marzipano.toJSON()
  expect(tree).toMatchSnapshot()
})
