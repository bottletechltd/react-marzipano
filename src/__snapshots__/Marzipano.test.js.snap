// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot 1`] = `
<div
  className=""
  style={Object {}}
/>
`;
